src/Loop.php

```diff
@@ -59,155 +59,162 @@ public static function stop()
     /**
      * Defer the execution of a callback.
      *
-     * @param callable $callback The callback to defer.
+     * @param callable(mixed $data, string $watcherIdentifier) $callback The callback to defer.
+     * @param mixed $data Arbitrary data given to the callback function as the $data parameter.
      *
-     * @return string An identifier that can be used to cancel, enable or disable the event.
+     * @return string An identifier that can be used to cancel, enable or disable the watcher.
      */
-    public static function defer(callable $callback)
+    public static function defer(callable $callback, $data = null)
     {
-        return self::get()->defer($callback);
+        return self::get()->defer($callback, $data);
     }
 
     /**
      * Delay the execution of a callback. The time delay is approximate and accuracy is not guaranteed.
      *
-     * @param callable $callback The callback to delay.
+     * @param callable(mixed $data, string $watcherIdentifier) $callback The callback to delay.
```

Comment:
You're right. I think it feels inconsistent to me, because of all the methods accepting `$watcherId` have it as first argument (`enable` and so on). They're not even connected, but it feels inconsistent, don't know why.